logstash-core/lib/logstash/agent.rb

```diff
@@ -75,9 +75,12 @@ def execute
       Stud.interval(@reload_interval) { reload_state! }
     else
       while !Stud.stop?
-        if clean_state? || running_pipelines?
-          sleep 0.5
-        else
+        if running_user_defined_pipelines?
+          sleep(0.5)
+        elsif running_pipelines?
+          logger.debug("Shutting down system pipelines")
+          shutdown_pipelines
+        elsif clean_state? || !running_pipelines?
```

Comment:
I think we do because we need to support the following usecase: 

1. Logstash + xpack monitoring
2. Logstash start with a finite pipeline ( AKA count => 2 )

When the pipeline defined by *2* is done  Logstash need to shutdown, Not signal is involved in that scenario?